src/Installer/redist-installer/targets/GenerateLayout.targets

```diff
@@ -35,8 +35,7 @@
       <PublicBaseURL Condition="$(PublicBaseURL.StartsWith('file:')) and '$(OS)' != 'Windows_NT'">$([System.Text.RegularExpressions.Regex]::Replace('$(PublicBaseURL)', '%28file:\/{1,}%29%28.+%29', 'file:///%242'))</PublicBaseURL>
 
       <NetRuntimeRid Condition="'$(NetRuntimeRid)' == ''">$(HostRid)</NetRuntimeRid>
-      <NetRuntimeRid Condition=" ('$(OSName)' == 'win' or '$(OSName)' == 'osx' or '$(OSName)' == 'freebsd') and '$(DotNetBuildSourceOnly)' != 'true' ">$(OSName)-$(Architecture)</NetRuntimeRid>
-      <NetRuntimeRid Condition="'$(DotNetBuild)' != 'true' and $(NetRuntimeRid.StartsWith('mariner.2.0'))">$(HostRid.Replace('mariner.2.0', 'cm.2'))</NetRuntimeRid>
+      <NetRuntimeRid Condition="$(NetRuntimeRid.StartsWith('mariner.2.0'))">$(NetRuntimeRid.Replace('mariner.2.0', 'cm.2'))</NetRuntimeRid>
```

Comment:
It was either adding explicitly adding illumos and solaris here, or simplifying the condition; I chose the latter. It's part of this PR?

Reply:
ah ok I didn't see the connection to the illumos change, and I still haven't fully convinced myself that removing these are really "redundant".

Is `$(HostRid)` really the same as `$(OSName)-$(Architecture)` in all cases?
Why can we remove the `'$(DotNetBuild)' != 'true'` check?

Reply:
I've set the default to `$(OSName)-$(Architecture)` as that is needed for cross-build.

>  Why can we remove the '$(DotNetBuild)' != 'true' check?

This predates the .NET 8 https://github.com/dotnet/sdk/commit/5b92cb03ea4ce9eae5760cf2193e74414a1ae4b2 and it's not needed because we are using base-os RIDs since .NET 8 (https://github.com/dotnet/docs/issues/35398).

Reply:
I still don't understand why we do that, looking at https://github.com/dotnet/sdk/blob/6ef345b1ff76eec2de6a0deeda321cae45da655b/src/Installer/redist-installer/targets/GetRuntimeInformation.targets#L5-L24 HostRid is potentially different.

Reply:
Lets talk in the main thread.